src/altinn-app-frontend/src/features/form/data/update/updateFormDataSagas.ts

```diff
@@ -86,6 +88,26 @@ function* runValidations(field: string, data: any, componentId: string | undefin
   );
 
   const componentValidations = validationResult?.validations[layoutId][componentId];
+
+  let customComponentValidations: IComponentValidations = {};
+  if (component.type === 'DatePicker') {
+    const fieldKey: keyof IDataModelBindings = 'simpleBinding';
+    const flagBasedMinDate = getFlagBasedDate(component.minDate as DateFlags) ?? component.minDate;
+    const flagBasedMaxDate = getFlagBasedDate(component.maxDate as DateFlags) ?? component.maxDate;
+    const datepickerValidations = validateDatepickerFormData(
+      data,
+      flagBasedMinDate,
+      flagBasedMaxDate,
+      component.format,
+      state.language.language,
+    );
+    customComponentValidations = {
+      [fieldKey]: datepickerValidations,
+    };
+  }
```

Comment:
Even though I really hope this thing is temporary until we take a look at #377, maybe move this out to a separate function in `validation.ts`? You could grab the opportunity to name this particular flavor of validations (are they immediate/instant/delayed?). The way I see it, validation hooks could possibly occur at several points in time:
1. When the user types/fills out something (possibly filtering/changing the value, such as with number formatting fields only accepting numbers? This should in turn give the user some kind of feedback - possibly through a short-lived validation message - see #288)
2. When the user has stopped typing (aka `useDelayedSaveState`), before/after the data is saved to the server
3. If the field is set up with `"triggers": ["validation"]` it should also ask the server for any validation messages(?) at this point
4. When page validation (or all-page validation) runs, all the previous validations should run, and in addition any required fields should produce validation messages if not properly filled out.

Reply:
Yeah I was also hoping this would be temporary 😅 not super happy about essentially duplicating this bit. I'll refactor a bit.